code/drasil-theory/Theory/Drasil/Theory.hs

```diff
@@ -70,6 +71,11 @@ instance HasReference       TheoryModel where getReferences = ref
 instance ConceptDomain      TheoryModel where cdom = cdom . view con
 -- | Finds any additional notes for the 'TheoryModel'.
 instance HasAdditionalNotes TheoryModel where getNotes = notes
+
+-- TODO: I think we should be gathering these from the ModelKinds of the TheoryModel.
+--       If we need "more than 1 ModelKind" in the TheoryModel, we may need to create 
+--       a "stacked model" that allows for composing them.
```

Comment:
If we choose to do something like this with a "MultiModel" of sorts, then the _defq, _invs, _dfun lists should likely be completely derived from the the models held within the "MultiModel". However, it may prove a bit difficult to work with for the above Theory typeclass as I don't see a clear path to creating a nice Lens for a "MultiModel".

Reply:
One nice thing about a "MultiModel" however, is that it would make TM definitions a bit shorter.

Reply:
Let's wait and see if/when this actually arises in practice.